nats-server-config-reloader.yaml

```diff
@@ -0,0 +1,114 @@
+package:
+  name: nats-server-config-reloader
+  version: "0.17.1"
+  epoch: 0
+  description: "NATS server configuration reloader utility"
+  copyright:
+    - license: Apache-2.0
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/nats-io/nack.git
+      tag: v${{package.version}}
+      expected-commit: 03e5e930036a5baacc12ef604cd57ad6ff6ada96
+
+  - uses: go/build
+    with:
+      packages: ./cmd/nats-server-config-reloader
+      output: nats-server-config-reloader
+      ldflags: -X main.BuildTime=$(date -u +%Y-%m-%dT%H:%M:%S%z) -X main.GitInfo=$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)$(git diff --quiet || echo "-dirty") -X main.Version=v${{package.version}}
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: "Compat package for ${{package.name}}"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/local/bin
+          ln -sf /usr/bin/nats-server-config-reloader ${{targets.contextdir}}/usr/local/bin/nats-server-config-reloader
+          ln -sf /usr/local/bin/nats-server-config-reloader ${{targets.contextdir}}/nats-server-config-reloader
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}
+      pipeline:
+        - runs: |
+            /usr/bin/nats-server-config-reloader --help
+            /usr/local/bin/nats-server-config-reloader --help
+            /nats-server-config-reloader --help
+
+  - name: ${{package.name}}-oci-entrypoint
+    description: Entrypoint for ${{package.name}}
+    pipeline:
+      - runs: |
+          install -m755 cicd/assets/entrypoint.sh ${{targets.contextdir}}/entrypoint.sh
+      - uses: strip
+    test:
+      pipeline:
+        - name: "Check expected scripts exist"
+          runs: |
+            stat /entrypoint.sh
+
+update:
+  enabled: true
+  github:
+    identifier: nats-io/nack
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - bash
+        - nats-server
+        - curl
+        - jq
+  pipeline:
+    - name: "Verify basic installation"
+      runs: |
+        nats-server-config-reloader --version 2>&1 | grep "${{package.version}}"
+        nats-server-config-reloader --help
+    - name: "Test nats-server-config-reloader with nats-server"
+      runs: |
+        echo "Starting NATS server with a configuration file..."
+        cat <<EOF > /tmp/nats-server.conf
+        pid_file: "/tmp/nats-server.pid"
+        http: 8222
+        EOF
+
+        nats-server -c /tmp/nats-server.conf -l /tmp/nats-server.log &
+        sleep 2
+
+        echo "Verifying NATS server process..."
+        if ! pgrep -f "nats-server -c /tmp/nats-server.conf"; then
+          echo "NATS server did not start properly!"
+          exit 1
+        fi
+
+        # Start the config reloader in the background
+        nats-server-config-reloader -config /tmp/nats-server.conf -pid /tmp/nats-server.pid &
```

Comment:
Let's use `test/daemon-check-output` here, run the `nats-server` in the `setup:` and ensure the config is updated in the `post:` steps.